website/pages/docs/subscriptions.mdx

```diff
@@ -0,0 +1,151 @@
+
+This guide covers how to implement subscriptions in GraphQL.js, when to use them, and what to consider in production environments.
+
+## What is a subscription?
+
+A subscription is a GraphQL operation that delivers ongoing results to the client when a specific event happens. Unlike queries or mutations, which return a single response, a subscription delivers data over time through a persistent connection.
+
+GraphQL.js implements the subscription execution algorithm, but it's up to you to connect it to your event system and transport layer. Most implementations use WebSockets for transport, though any full-duplex protocol can work.
+
+## How execution works
+
+The core of subscription execution in GraphQL.js is the `subscribe` function. It works similarly to `graphql()`, but returns an `AsyncIterable` of execution results
+instead of a single response:
+
+```js
+import { subscribe, parse } from 'graphql';
+import schema from './schema.js';
+
+const document = parse(`
+  subscription {
+    messageSent
+  }
+`);
+
+const iterator = await subscribe({ schema, document });
+
+for await (const result of iterator) {
+  console.log(result);
+}
+```
+
+Each time your application publishes a new `messageSent` event, the iterator emits a new result. It's up to your transport layer to manage the connection and forward these updates to the client.
+
+## When to use subscriptions
+
+Subscriptions are helpful when your application needs to respond to real-time events. For example:
+
+- Receiving new messages in a chat
+- Updating a live feed or activity stream
+- Displaying presence indicators (e.g., "user is online")
+- Reflecting real-time price changes
+
+If real-time delivery isn’t essential, consider using polling with queries instead. Subscriptions require more infrastructure and introduce additional complexity, especially around scaling and connection management.
+
+## Implementing subscriptions in GraphQL.js
+
+GraphQL.js supports subscription execution, but you’re responsible for setting up the transport and event system. At a minimum, you’ll need:
+
+- A `Subscription` root type in your schema
+- A `subscribe` resolver that returns an `AsyncIterable`
+- An event-publishing mechanism
+- A transport layer to maintain the connection
+
+The following examples use the [`graphql-subscriptions`](https://github.com/apollographql/graphql-subscriptions) package to set up an in-memory event system.
+
+### Install dependencies
+
+Start by installing the necessary packages:
+
+```bash
+npm install graphql graphql-subscriptions
+```
+
+To serve subscriptions over a network, you’ll also need a transport implementation. One option is [`graphql-ws`](https://github.com/enisdenjo/graphql-ws), a community-maintained WebSocket library. This guide focuses on schema-level implementation.
+
+### Set up a pub/sub instance
+
+Create a `PubSub` instance to manage your in-memory event system:
+
+```js
+import { PubSub } from 'graphql-subscriptions';
```

Comment:
Besides this unmaintained library, we have an alternative package `@graphql-yoga/subscriptions` that uses the standard [`EventTarget`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget), and it also allows you to use other event sources like Redis.
```ts
import { createPubSub } from '@graphql-yoga/subscriptions'
import { Redis } from 'ioredis'
import { createRedisEventTarget } from '@graphql-yoga/redis-event-target'
 
const publishClient = new Redis()
const subscribeClient = new Redis()
 
const eventTarget = createRedisEventTarget({
  publishClient,
  subscribeClient
})
 
const pubSub = createPubSub({ eventTarget })
```